### ISSUE TITLE
Cannot set Container code to ON

### ISSUE THREAD

**Issue by bnckamaris** [duracelltomi/gtm4wp#302]
When i place the GTM ID and saving the setings, the plugin switch the Container code to automaticaly to off

Please check this video: https://www.loom.com/share/9170f397e6f5459ebff716093f448ac6

Can anyone help ?

**Comment by duracelltomi** [comment]
I've got similar reports but I was not able to reproduce this on my end.
Do you have any steps to do in order to see this bug?

**Comment by alimdi** [comment]
Same Issue here with 2 warning messages :

Warning: Undefined array key "container-on" in /home/clients/e02236d5a95afc7cada35eeedcfcd100/sites/XXX.YY/wp-content/plugins/duracelltomi-google-tag-manager/admin/admin.php on line 659

Warning: Undefined array key "compat-mode" in /home/clients/e02236d5a95afc7cada35eeedcfcd100/sites/XXX.YY/wp-content/plugins/duracelltomi-google-tag-manager/admin/admin.php on line 660

Thank you for your help

**Comment by jcagroenen** [comment]
@duracelltomi 

Im also getting this, but locally fixed this by doing:

            $container_on_off = isset($options['container-on']) ? (bool) $options['container-on'] : false;
            $container_compat = isset($options['compat-mode']) ? (int) $options['compat-mode'] : 0;


can you please look into this?

**Comment by dangelion** [comment]
Same error here. Any solution?

**Comment by duracelltomi** [comment]
You're right to keep pushing on this — thanks to everyone who added detail (and @jcagroenen for the `isset()` diagnosis). Root cause in 1.x: on a brand-new install the `gtm4wp-options` row doesn't exist yet, and the admin read `$options['container-on']` / `$options['compat-mode']` without guarding for the missing keys, so the toggle wouldn't stick.

In the 2.0 rewrite this is resolved by design: the separate on/off and compatibility-mode switches are gone — container output is now driven by the per-container table, with an explicit "Off" in the placement setting — and options are always loaded through a service that merges module defaults over the stored row, so a missing/empty option row can no longer produce an undefined-key warning or a reverting toggle. The hard-coded `GTM4WP_HARDCODED_*` / Composer-install path (@ehclau) is handled explicitly too.

Ships in 2.0 beta1 — closing as resolved. Thanks all!
